enhancements/cluster-logging/azure_workload_identity_auth.md

```diff
@@ -0,0 +1,264 @@
+---
+title: workload_identity_support_for_azure_monitor_logs
+authors:
+  - "@calee"
+reviewers:
+  - "@jcantrill"
+  - "@alanconway"
+approvers:
+  - "@jcantrill"
+  - "@alanconway"
+api-approvers:
+  - "@jcantrill"
+  - "@alanconway"
+creation-date: 2025-04-30
+last-updated: 2025-04-30
+status: implementable
+tracking-link:
+  - https://issues.redhat.com/browse/LOG-4782
+---
+
+# Workload Identity Support for Azure Monitor Logs
+
+## Release Sign-off Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+[Azure Monitor Logs](https://learn.microsoft.com/en-us/azure/azure-monitor/logs/data-platform-logs) is a comprehensive service provided by Microsoft Azure that enables the collection, analysis, and actioning of telemetry data across various Azure and on-premises resources.
+
+This proposal enhances the Azure Monitor Logs integration by implementing secure, short-lived authentication with [Microsoft Entra Workload Identity (WID)](https://learn.microsoft.com/en-us/entra/workload-id/workload-identities-overview) through federated tokens. The update will leverage a pending, upstream Vector PR, [azure_logs_ingestion feature](https://github.com/vectordotdev/vector/pull/22912), that will utilize the new Log Ingestion API.
+
+## Motivation
+
+The current Azure Monitor Logs integration relies on long-lived credentials ([shared_key](https://learn.microsoft.com/en-us/previous-versions/azure/azure-monitor/logs/data-collector-api?tabs=powershell#authorization)) and a deprecated [API](https://learn.microsoft.com/en-us/previous-versions/azure/azure-monitor/logs/data-collector-api), which poses potential security risks. By adopting Microsoft Entra Workload Identity (WID), this enhancement ensures secure, short-lived credential access to Azure Monitor Logs while removing reliance on a deprecated, soon to be retired, API.
+
+### User Stories
+
+- As an administrator, I want to be able to forward logs from my OpenShift cluster to Azure Monitor Logs using federated tokens, removing the need for long-lived, static credentials.
+
+### Goals
+
+- Enable Vector's Azure Monitor Logs sink to authenticate with short-lived federated token credentials.
+
+### Non-Goals
+
+- Supporting authentication other than short lived federated token credentials.
```

Comment:
one other, shared_key.   We need to ensure long-lived continues to be supported as well.